wolfcrypt/src/asn.c

```diff
@@ -11030,15 +11075,127 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
+                ret = ASN_PARSE_E;
+        }
+
+        if (ret == 0) {
+            GetInteger7Bit(input, inOutIdx, inSz);
+            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
+                ret = ASN_PARSE_E;
+        }
+        if (ret == 0) {
+            SkipObjectId(input, inOutIdx, inSz);
+            ret = ASNToHexString(input, inOutIdx, (char**)&params->prime, inSz,
+                                                 NULL, DYNAMIC_TYPE_ECC_BUFFER);
+        }
+        if (ret == 0) {
+            params->size = (int)XSTRLEN(params->prime) / 2;
+
+            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
+                ret = ASN_PARSE_E;
+        }
+        if (ret == 0) {
+            ret = ASNToHexString(input, inOutIdx, (char**)&params->Af, inSz,
+                                                 NULL, DYNAMIC_TYPE_ECC_BUFFER);
+        }
+        if (ret == 0) {
+            ret = ASNToHexString(input, inOutIdx, (char**)&params->Bf, inSz,
+                                                 NULL, DYNAMIC_TYPE_ECC_BUFFER);
+        }
+        if (ret == 0) {
+            if (input[*inOutIdx] == ASN_BIT_STRING) {
+                len = 0;
+                ret = GetASNHeader(input, ASN_BIT_STRING, inOutIdx, &len, inSz);
+                inOutIdx += len;
+            }
+        }
+        if (ret == 0) {
+            ret = ASNToHexString(input, inOutIdx, (char**)&point, inSz, NULL,
+                                                       DYNAMIC_TYPE_ECC_BUFFER);
+        }
+        if (ret == 0) {
+            params->Gx = XMALLOC(params->size * 2 + 2, NULL,
+                                                       DYNAMIC_TYPE_ECC_BUFFER);
+            params->Gy = XMALLOC(params->size * 2 + 2, NULL,
+                                                       DYNAMIC_TYPE_ECC_BUFFER);
+            if (params->Gx == NULL || params->Gy == NULL)
+                ret = MEMORY_E;
+        }
+        if (ret == 0) {
+            XMEMCPY((char*)params->Gx, point + 2, params->size * 2);
+            XMEMCPY((char*)params->Gy, point + params->size * 2 + 2,
+                                                              params->size * 2);
+            ((char*)params->Gx)[params->size * 2] = '\0';
+            ((char*)params->Gy)[params->size * 2] = '\0';
+            XFREE(point, NULL, DYNAMIC_TYPE_ECC_BUFFER);
+            ret = ASNToHexString(input, inOutIdx, (char**)&params->order, inSz,
+                                                 NULL, DYNAMIC_TYPE_ECC_BUFFER);
+        }
+        if (ret == 0) {
+            params->cofactor = GetInteger7Bit(input, inOutIdx, inSz);
+
+            params->oid = NULL;
+            params->oidSz = 0;
+            params->oidSum = 0;
+
+            if (wc_ecc_set_custom_curve(key, params) < 0) {
+                ret = ASN_PARSE_E;
+            }
+            key->deallocSet = 1;
+            params = NULL;
+        }
+        if (params != NULL) {
```

Comment:
Recommend adding ECC local function for "freeing" and DP (ecc_set_type) so it can be called here and in `wc_ecc_free`. Ex: `WOLFSSL_LOCAL void wc_ecc_free_dp(ecc_set_type* dp, void* heap);`

Also make sure and add the key->heap arg.

Reply:
Fixed